### ISSUE TITLE
Weird thing when compare Mask R-CNN and Cascade Mask R-CNN

### ISSUE THREAD

**Issue by TuanNguyenSKKU** [open-mmlab/mmdetection#3927]
hello mmdetection team and everyone,

When I trained the model with a custom dataset (~1000 images) for two methods: A (Mask R-CNN ResNeXt-101-32x4d) and B (Cascade Mask R-CNN ResNet-50). The ratio for train:val:test is 70:15:15.

- For training and validation: the precision and recall of B is better than A.
- For testing: the precision and recall of A is better than B (for almost IoU).
![Picture1](https://user-images.githubusercontent.com/55343855/95952565-51987900-0e33-11eb-9fcb-5759d43517fd.png)

The testing images are captured with different backgrounds compared to train/val set.

Please help me explain what is the reason? And what do I need to do to make it more correct?
I hope to receive your support.
Thank you.

**Comment by shinya7y** [comment]
The dataset is small, and the difference does not seem to be statistically significant.
Training models with 5-10 different random seeds might make the results reliable.
However, adding data augmentation will be more important than comparing A with B.

**Comment by yhcao6** [comment]
The first row compares their mAP from 0.5:0.95, but the second row only compares mAP of 0.9, so it is not comparable.
Besides, your dataset is too small so the results may have some fluctuation. To be more accurate you can do several experiments and take the average.

**Comment by TuanNguyenSKKU** [comment]
Thank you for your suggestions.

**Comment by TuanNguyenSKKU** [comment]
For COCO, PASCAL VOC, KITTI,... dataset, Cascade Mask R-CNN give better scores than Mask R-CNN at the same architectures.

But when I tried much time with my custom dataset (using pre-trained COCO), Mask R-CNN always give better scores than Cascade Mask R-CNN (using the same ResNeXt-101-32x4d) on the testing set (sometimes on the validation set).

Can you explain more about this phenomenon?
Thank you.
